lib/matplotlib/_mathtext.py

```diff
@@ -128,50 +129,59 @@ def __init__(self, box: Box):
-                    *[y1 for x1, y1, x2, y2 in self.rects], 0]) - 1
+                    *[x1 for x1, y1, x2, y2 in self.rects], 0])
+        ymin = min([*[oy - max(info.metrics.ymax, info.metrics.iceberg)
+                      for ox, oy, info in self.glyphs],
+                    *[y1 for x1, y1, x2, y2 in self.rects], 0])
         xmax = max([*[ox + info.metrics.xmax for ox, oy, info in self.glyphs],
-                    *[x2 for x1, y1, x2, y2 in self.rects], 0]) + 1
+                    *[x2 for x1, y1, x2, y2 in self.rects], 0])
         ymax = max([*[oy - info.metrics.ymin for ox, oy, info in self.glyphs],
-                    *[y2 for x1, y1, x2, y2 in self.rects], 0]) + 1
+                    *[y2 for x1, y1, x2, y2 in self.rects], 0])
+        # Rasterizing can leak into the neighboring pixel, hence the +/-1; it
+        # will be cropped at the end.
+        xmin = math.floor(xmin - 1)
+        ymin = math.floor(ymin - 1)
+        xmax = math.ceil(xmax + 1)
+        ymax = math.ceil(ymax + 1)
         w = xmax - xmin
-        h = ymax - ymin - self.box.depth
-        d = ymax - ymin - self.box.height
-        image = FT2Image(int(np.ceil(w)), int(np.ceil(h + max(d, 0))))
+        h = max(-ymin, 0)
+        d = max(ymax, 0)
+        image = FT2Image(w, h + d)
 
-        # Ideally, we could just use self.glyphs and self.rects here, shifting
-        # their coordinates by (-xmin, -ymin), but this yields slightly
-        # different results due to floating point slop; shipping twice is the
-        # old approach and keeps baseline images backcompat.
-        shifted = ship(self.box, (-xmin, -ymin))
-
-        for ox, oy, info in shifted.glyphs:
+        for ox, oy, info in self.glyphs:
+            ox -= xmin
+            oy -= (-h + info.offset)
             info.font.draw_glyph_to_bitmap(
-                image,
-                int(ox),
-                int(oy - np.ceil(info.metrics.iceberg)),
-                info.glyph,
-                antialiased=antialiased)
-        for x1, y1, x2, y2 in shifted.rects:
+                image, ox, oy, info.glyph, antialiased=antialiased)
+        for x1, y1, x2, y2 in self.rects:
+            x1 -= xmin
+            x2 -= xmin
+            y1 -= -h
+            y2 -= -h
             height = max(int(y2 - y1) - 1, 0)
             if height == 0:
                 center = (y2 + y1) / 2
                 y = int(center - (height + 1) / 2)
             else:
                 y = int(y1)
-            image.draw_rect_filled(int(x1), y, int(np.ceil(x2)), y + height)
-        return RasterParse(0, 0, w, h + d, d, image)
+            image.draw_rect_filled(int(x1), y, math.ceil(x2), y + height)
+
+        image = np.asarray(image)
+        while h and (image[0] == 0).all():
+            image = image[1:]
+        while d and (image[-1] == 0).all():
+            image = image[:-1]
+        return RasterParse(xmin, 0, w, h + d, d, image)
```

Comment:
@anntzer `h` and `d` don't change in these loops; are they supposed to be reduced each time?

Reply:
Yes, sorry about that, I had noticed this earlier and force-pushed a version of the mt1 branch where I am also more careful with positioning the origin of the rasterized image relative to the origin of the text string, can you try again?